### DISCUSSION TITLE
IR remote support for ESP32C2

### DISCUSSION THREAD

**Discussion by ferbulous** [arendst/Tasmota#19796] [Support request] [answered]
Hi, I'm trying to add ir remote for esp32c2,  ```#define USE_IR_REMOTE_FULL```
But I'm getting this error

I'm aware this is new chip for tasmota support, so I was wondering if there's a way to fix so it compiles?

```
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\hal\lv_hal_disp.c.o' from cache
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\hal\lv_hal_indev.c.o' from cache
C:/Users/nurul/Documents/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_05_irremote_full.ino:62:10: fatal error: IRremoteESP8266.h: No such file or directory

*************************************************************************
* Looking for IRremoteESP8266.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:IRremoteESP8266.h"
* Web  > https://registry.platformio.org/search?q=header:IRremoteESP8266.h
*
*************************************************************************

   62 | #include <IRremoteESP8266.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\hal\lv_hal_tick.c.o' from cache
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\misc\lv_anim.c.o' from cache
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\misc\lv_anim_timeline.c.o' from cache
Retrieved `.pio\build\tasmota32c2-arduino30\lib917\lvgl\misc\lv_area.c.o' from cache
*** [.pio\build\tasmota32c2-arduino30\src\tasmota.ino.cpp.o] Error 1
=========================================================== [FAILED] Took 295.55 seconds ===========================================================

Environment            Status    Duration
---------------------  --------  ------------
tasmota32c2-arduino30  FAILED    00:04:55.554
====================================================== 1 failed, 0 succeeded in 00:04:55.554 ======================================================
```

**Comment by Jason2866** [comment, marked as answer]
IRremote is disabled since the used library does not work with Arduino 3.0. There is no quick fix. The Timer API has changed and our tries to adopt to the new API does compile but has horrible recognition rates.
Probably takes some time. The maintainer of the lib is knowing the problem but is short on free time to work on it.

**Reply by ferbulous** [reply]
Thanks, guess I'll have to wait until it's fixed on the next build

**Reply by barbudor** [reply]
You'll save time by using one of the already supported CPU.
Why going for a C2 which support is very preliminary?

**Comment by Jason2866** [comment]
To be very clear we are going to add official Tasmota support for C6 some day.
The C2 build is highly! experimental and it could be there is never an official Tasmota version for.
Reason: C2 will get no official Arduino support. The Arduino version we use for the C2 is done by Tasmota crew. If the maintaining Arduino support for the C2 is getting to complicated / time intensive i have to drop.

100% sure the C2 will never have all features of the other ESP32 MCU. The resources of the C2 are too limited.

**Reply by lboue** [reply]
Just for information, for those who are interested. ESP32-C2 is available:

- https://eu.mouser.com/ProductDetail/Espressif-Systems/ESP8684-DevKitM-1-H4
- https://www.aliexpress.com/item/1005006023619761.html

**Comment by ferbulous** [comment]
Closed, IR can now be compiled succesfully
